EuPathDB/src/main/resources/assets/admin/admin-ui.js

```diff
@@ -0,0 +1,332 @@
+}
+
+function loadUserAssignment(groupId) {
+  // load groups; once loaded, choose the passed group if present
+  loadGroupPicker(() => {
+    if (groupId) {
+      $('#groupPicker option[value="' + groupId + '"]').prop('selected', true);
+    }
+  });
+}
+
+function initNewSubscriptionForm() {
+  $("#title").html("Add New Subscription");
+  $("#mode").val("new");
+  useEditPanel();
+}
+
+function loadSubscription(id) {
+  doGet("/oauth/subscriptions/" + id, sub => {
+    $("#title").html("Subscription: " + sub.displayName);
+    useDisplayPanel();
+
+    // fill display area
+    $("#subscriptionId").html(sub.subscriptionId);
+    $("#isActive").html(sub.isActive ? "yes" : "no");
+    $("#groups").html(sub.groups.map(group =>
+        '<li><a href="/oauth/assets/admin/group.html?id=' + group.groupId + '">' + group.displayName + '(' + group.groupId + ')</a></li>'
+    ));
+
+    // fill form
+    $("#mode").val("edit");
+    $("#cancelButton").show();
+    $("#displayNameInput").val(sub.displayName);
+    let selectedValue = sub.isActive ? "yes" : "no";
+    $('#isActiveInput option[value="' + selectedValue + '"]').prop('selected', true);
+    
+  });
+}
+
+function saveSubscription() {
+  // gather data
+  var isNew = $("#mode").val() == "new";
+  var data = {
+    "displayName": $("#displayNameInput").val(),
+    "isActive": $("#isActiveInput")[0].selectedOptions[0].value == "yes"
+  };
+  if (isNew) {
+    doPost("/oauth/subscriptions", data, response => {
+      visitSubscription(response.subscriptionId);
+    });
+  }
+  else {
+    var subscriptionId = $("#subscriptionId").html();
+    doPost("/oauth/subscriptions/" + subscriptionId, data, () => {
+      visitSubscription(subscriptionId);
+    });
+  }
+}
+
+function initloadNewGroupForm() {
+  $("#title").html("Add New Group");
+  $("#mode").val("new");
+  useEditPanel();
+  loadSubscriptionPicker();
+}
+
+function loadGroup(id) {
+  const userArrayToHtml = users => users.map(user => "<li>" + user.userId + ": " + user.name + " (" + user.organization + ")</li>");
+  doGet("/oauth/groups/" + id, group => {
+
+    // load subscriptions; once loaded, display this group's subscription name and select it in the drop-down
+    loadSubscriptionPicker(() => {
+      $('#subscriptionPicker option[value="' + group.subscriptionId + '"]').prop('selected', true);
+      let sub = globalState.subscriptionMeta.filter(sub => sub.subscriptionId == group.subscriptionId)[0];
+      $("#subscriptionName").html('<a href="/oauth/assets/admin/subscription.html?id=' + sub.subscriptionId + '">' + sub.displayName + "</a> (" + (sub.isActive ? "active" : "inactive") + ")");
+    });
+
+    $("#title").html("Group: " + group.displayName);
+    useDisplayPanel();
+
+    // fill display area
+    $("#groupId").html(group.groupId);
+    $("#subscriptionToken").html(group.subscriptionToken);
+    $("#leads").html(userArrayToHtml(group.leadUsers));
+    $("#members").html(userArrayToHtml(group.members));
```

Comment:
These need to be HTML for the list element parts, but you can escape HTML when interpolating database-derived data in `userArrayToHtml`.

Alternatively build the list formally with DOM methods and use `.text()` for user content.